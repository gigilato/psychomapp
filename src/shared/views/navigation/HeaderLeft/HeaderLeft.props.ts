import { HeaderBackButtonProps } from '@react-navigation/native-stack/lib/typescript/src/types'

export interface IHeaderLeftProps extends HeaderBackButtonProps {
  type?: 'back' | 'close'
  onPress?: () => void
  restoreTabBar?: boolean
}
