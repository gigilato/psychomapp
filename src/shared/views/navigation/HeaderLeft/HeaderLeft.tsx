import { router } from 'expo-router'

import { setTabBarVisibility } from '$infra/layout'
import { PressableIcon } from '$molecules'
import { IHeaderLeftProps } from '$navigation/HeaderLeft/HeaderLeft.props'

export const HeaderLeft = ({
  canGoBack,
  onPress,
  restoreTabBar = false,
  type = 'back',
  ...props
}: IHeaderLeftProps) => {
  if (!canGoBack) return null
  return (
    <PressableIcon
      ID="goback"
      icon={type === 'back' ? 'chevron-left' : 'x'}
      iconSize={30}
      onPress={() => {
        onPress?.()
        if (restoreTabBar) setTabBarVisibility(true)
        router.back()
      }}
    />
  )
}
