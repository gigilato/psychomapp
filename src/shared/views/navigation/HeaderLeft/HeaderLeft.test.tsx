import { render } from '@testing-library/react-native'

import { HeaderLeft } from './HeaderLeft'

describe('HeaderLeft', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<HeaderLeft canGoBack />)
    expect(toJSON()).toMatchSnapshot()
  })
})
