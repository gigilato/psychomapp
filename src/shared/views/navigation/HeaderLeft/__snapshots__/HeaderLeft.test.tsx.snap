// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLeft renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={5}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID="goback"
>
  <Text />
</View>
`;
